adr/0013-suma-integration.md

```diff
@@ -0,0 +1,141 @@
+# 13. SUMA Integration
+
+Date: 2024-05-07
+
+## Status
+
+Accepted
+
+## Context
+
+Keeping SAP infrastructure up-to-date and secure is a crucial task for system administrators, allowing vulnerabilities reduction and overall system improvement in terms of features and performance.
+
+In order for corrective actions to be taken in a timely manner, Trento aims to provide visibility about *Available Software Updates* in registered SAP hosts.
+
+> *Available Software Updates* can be **security patches for existing CVEs**, **bug fixes** or **software enhancements** (*Relevant Patches*) which, when applied, may require upgrading one or more installed packages (*Upgradable Packages*) in the host.
+
+[SUSE Manager](https://www.suse.com/products/suse-manager/) (**SUMA** from here onwards), is SUSE's product that already provides *infrastructure management* and specifically *automated software updates and patches* solutions while also [exposing APIs](https://www.suse.com/c/suse-manager-and-the-rest-api/) to enhance automation and facilitate integration with external tools.
+
+These premises make it both **strategically and technically convenient** leveraging SUMA's capabilities within Trento to provide a comprehensive SUSE driven experience to SAP userbase.
+
+> Current integration only focuses on enhancing visibility by running software updates discoveries. No corrective actions are taken on the hosts by Trento.
+>
+> SUMA version, and related HTTP APIs, currently targeter for integration is [4.3](https://documentation.suse.com/suma/4.3/api/suse-manager/index.html)
+
+## Considered Options
+
+- **Eventless Integration**: just proxying SUMA API calls directly within Trento's APIs, without the need of interacting with Trento's event driven nature
```

Comment:
I think that here, on top of the table, we had 2 real scenario, as the health usage on the host health was initially taken as granted, as it was a business requirement.

1. Query and show data coming directly from SUMA (with the disadvantage that the data coming might not be consistent with the host health)
2. Query and store data coming from SUMA, so the health of the host is calculated always with the latest discovered suma info

With that being said, I would said that this initial implementation goes with the 2nd option, as we don't have any capability by now to know if there are changes in SUMA without polling directly.
With this, I would say that the 1st option could be "workarounded" just dispatching the health updated command everytime we query SUMA, but we discarded for not adding this side effect in "read" api query